packages/compiler-cli/src/ngtsc/annotations/component/src/handler.ts

```diff
@@ -1483,23 +1471,77 @@ export class ComponentDecoratorHandler implements
 
       // Keep track of how this class made it into the current source file
       // (which ts.ImportDeclaration was used for this symbol).
-      resolutionData.deferrableDeclToImportDecl.set(
-          decl.node as unknown as Expression, imp.node as unknown as Expression);
+      resolutionData.deferrableDeclToImportDecl.set(decl.node, imp.node);
 
       this.deferredSymbolTracker.markAsDeferrableCandidate(
           node, imp.node, componentClassDecl, isDeferredImport);
     }
   }
 
-  /** Resolves the triggers of the defer block to the elements that they're pointing to. */
-  private resolveDeferTriggers(
-      block: TmplAstDeferredBlock, triggers: TmplAstDeferredBlockTriggers,
-      componentBoundTarget: BoundTarget<DirectiveMeta>,
-      triggerElements: Map<TmplAstDeferredTrigger, TmplAstElement|null>): void {
-    Object.keys(triggers).forEach(key => {
-      const trigger = triggers[key as keyof TmplAstDeferredBlockTriggers]!;
-      triggerElements.set(trigger, componentBoundTarget.getDeferredTriggerTarget(block, trigger));
-    });
+  private compileDeferBlocks(resolution: Readonly<Partial<ComponentResolutionData>>):
+      R3DeferMetadata {
+    if (resolution.deferBlockDepsEmitMode === DeferBlockDepsEmitMode.PerBlock) {
+      if (!resolution.deferBlockDependencies) {
+        throw new Error('deferBlockDependencies must be present when compiling in PerBlock mode');
```

Comment:
nit: we can mention here (and in a similar `throw`s within this function) that this is an internal error (i.e. unrelated to developers code):
```suggestion
        throw new Error('Internal error: deferBlockDependencies must be present when compiling in PerBlock mode');
```